Release/0.1.0

lyndsysimon opened the pull request.

lyndsysimon committed:
Refactor API for consistency

lyndsysimon committed:
Add Travis-CI integration

lyndsysimon committed:
Update README

lyndsysimon committed:
Update .travis.yml

lyndsysimon committed:
Add Travis badge

lyndsysimon committed:
Add linter

lyndsysimon committed:
Update version

pramttl commented on tests/test_explicit_keys.py:
```diff
@@ -23,23 +23,23 @@ class Delta(Charlie):
 
 
 def test_tracked_descendants():
```
(Optional) Could be updated to `test_progeny` to match the property method name. Generally on the pbm codebase we try to keep the test names something like `test_method_name` where `method_name` is the method being tested.

lyndsysimon replied:
Yeah, this test name should be updated.

pramttl commented on tests/test_simple_hierarchy.py:
```diff
@@ -18,29 +18,29 @@ class Delta(Charlie):
 
 
 def test_tracked_descendants():
```
Isn't this a duplicate test ? (Also present above in, test_explicit_keys.py)

lyndsysimon replied:
Not duplicate, no - the setup is different, as are the assertions. They're testing the different results of the same method under different circumstances.

pramttl commented on tests/test_tracking_flag.py:
```diff
@@ -22,18 +22,18 @@ class Delta(Charlie):
 
 
 def test_tracked_descendants():
```
Another duplicate ?

lyndsysimon committed:
move lookup methods to ProgenySet

lyndsysimon committed:
Update test names

rouge8 commented on src/progeny/__init__.py:
```diff
@@ -1,6 +1,22 @@
 import six
 
 
+class ProgenySet(set):
```
Implement `collections.Set` maybe? https://docs.python.org/3/library/collections.abc.html or at a minimum, this should probably be a frozenset (and if we do freeze it, test(s) to verify it's frozen 😃)

rouge8 commented on .travis.yml:
```diff
@@ -0,0 +1,13 @@
+  - "3.6"
+install:
+  - pip install tox-travis pycodestyle
+before_script:
```
@lyndsysimon I might push out a PR tweaking the Travis and tox setup

rouge8 replied:
https://github.com/GoodRx/progeny/pull/4

rouge8 commented on src/progeny/__init__.py:
```diff
@@ -17,31 +33,25 @@ def __init__(cls, name, bases, dct):
             cls.__progeny_key__ = cls._get_progeny_key()
 
+    @property
+    def progeny(self):
```
would progeny classes (e.g. `Alpha`, `Bravo`, etc. in the tests) also have a `.progeny`?

rouge8 replied:
to answer my own question: they do, which I don't love. I'm not sure if it'd be more confusing if they didn't have it nor how much effort. no strong opinion (and definitely not a blocker), just food for thought

lyndsysimon replied:
They should, yes. That behavior is not explicitly tested so I'm not going to promise that it works, but that was my intention.

rouge8 reviewed (commented):
Let's see if I understand this, adapting it to our `Backend` model:

```python
class Backend(ProgenyBase):
    def _get_progeny_key(cls):
        return cls.id

class Backend1(Backend):
    id = 'ONE'

class Backend2(Backend):
    id = 'TWO'
```

@lyndsysimon that look right?

rouge8 committed:
Use flake8 over pycodestyle and revamp Travis setup

rouge8 committed:
Run pytest with coverage

rouge8 committed:
Use isort to standardize imports

lyndsysimon committed:
Merge pull request #4 from GoodRx/tooling

Travis/tox setup

rouge8 committed:
.progeny is now dict-like instead of set-like

This patch updates the `.progeny` property to return an immutable
dict-like object rather than a set-like object with the descendents. The
main API difference is iteration: iterating over `.progeny` now iterates
over `__progeny_key__` rather than the subclasses.

rouge8 commented on README.rst:
```diff
@@ -53,10 +59,10 @@ In some cases, it may be useful to prevent descendant classes from being visible
 .. code:: python
 
-    from progeny import ProgenyBase
+    from progeny import progeny.Base
```
bad find/replace here

rouge8 commented on README.rst:
```diff
@@ -43,7 +49,7 @@ Now we can iterate over all of the subclasses of ``NotificationHandler``:
 
     def send_newsletter():
-        for handler in NotificationHandler.tracked_descendants():
+        for handler in NotificationHandler.progeny:
```
`.progeny.values()`

rouge8 reviewed (changes requested):
Two README fixes that could easily have been my fault 🙈

lyndsysimon committed:
Prepare for publishing to PyPI

lyndsysimon committed:
increment version
